exchanges/binance/binance.go

```diff
@@ -185,10 +186,12 @@ func (b *Binance) GetAggregatedTrades(arg *AggregatedTradeRequestParams) ([]Aggr
 	if !arg.EndTime.IsZero() {
 		params.Set("endTime", strconv.FormatInt(convert.UnixMillis(arg.EndTime), 10))
 	}
-	// If startTime and endTime are sent, time between startTime and endTime must be less than 1 hour.
-	if arg.Limit == 0 && !arg.StartTime.IsZero() && !arg.EndTime.IsZero() && arg.EndTime.Sub(arg.StartTime) > time.Hour {
+	// Fall back to batch requests
+	// if the requested limit is not supported by exchange or
+	// if startTime and endTime are set and time between startTime and endTime is more than 1 hour
+	if !limitSupported || (!arg.StartTime.IsZero() && !arg.EndTime.IsZero() && arg.EndTime.Sub(arg.StartTime) > time.Hour) {
 		// Split the request into multiple
-		return batchAggregateTrades(b, arg, params)
+		return b.batchAggregateTrades(arg, params)
```

Comment:
If I change my limit up to 1001 I get no error and no data, I know this is somewhat intentional but it breaks from what I am expecting. I know its a pain but we may need a check for unset startTime as well as an incorrect limit to return an error? 
```go
_, err := b.GetAggregatedTrades(&AggregatedTradeRequestParams{
		Symbol: currency.NewPair(currency.BTC, currency.USDT).String(),
		Limit:  1001,
	})
	if err != nil {
		t.Error("Binance GetAggregatedTrades() error", err)
	}
```
Maybe scoped here we can do a start time check and return the error: https://github.com/thrasher-corp/gocryptotrader/pull/588/files/a4341f513e6fceb0fb991c5e1194708c0328b2e8..b7214a1d514795754d543472e254e8fccae3c003#diff-8ff10965769d44e4c1fffd1a0d55ccf0bc16b1bdcce3b2cb7bea45fc4548a27fR174

Let me know what you think. Maybe I am misinterpreting things a little bit.

Reply:
start time is being set to 0 since arg.StartTime is empty n then u add 1 hr to that for endTime and no results coming back during that period

Reply:
Added some more tests as well